File: Data Structures & Algorithms/concatenation-of-array/submission-3.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> ans = nums;
        ans.resize(nums.size()*2);
        for(int i = 0; i < nums.size(); i++){
            ans[i + nums.size()] = nums[i];
        }
        return ans;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sortedNums  = []
        for i, num in enumerate(nums):
            sortedNums.append([num, i])
        sortedNums.sort()

        i = 0 #first index
        j = len(nums) - 1 # last index

        while i < j:
            if sortedNums[i][0] + sortedNums[j][0] == target:
                return [min(sortedNums[i][1],sortedNums[j][1]),
                max(sortedNums[i][1],sortedNums[j][1])]
            elif sortedNums[i][0] + sortedNums[j][0] < target:
                i += 1
            elif sortedNums[i][0] + sortedNums[j][0] > target:
                j -= 1
        return []
        
